crates/bevy_audio/src/audio.rs

```diff
@@ -393,3 +180,100 @@ impl GlobalVolume {
         }
     }
 }
+
+/// Bundle for playing a sound.
+///
+/// Insert this bundle onto an entity to trigger a sound source to begin playing.
+///
+/// If the handle refers to an unavailable asset (such as if it has not finished loading yet),
+/// the audio will not begin playing immediately. The audio will play when the asset is ready.
+///
+/// When Bevy begins the audio playback, an [`AudioSink`][crate::AudioSink] component will be
+/// added to the entity. You can use that component to control the audio settings during playback.
+#[derive(Bundle, Default)]
+pub struct AudioBundle<Source = AudioSource>
+where
+    Source: Asset + Decodable,
+{
+    /// Asset containing the audio data to play.
+    pub source: Handle<Source>,
+    /// Initial settings that the audio starts playing with.
+    /// If you would like to control the audio while it is playing,
+    /// query for the [`AudioSink`][crate::AudioSink] component.
+    /// Changes to this component will *not* be applied to already-playing audio.
+    pub settings: PlaybackSettings,
+}
+
+impl AudioBundle<AudioSource> {
+    /// Create an [`AudioBundle`] from a standard Bevy audio source.
+    ///
+    /// Use this if you are loading an audio file asset in the formats supported by Bevy.
+    pub fn from_audio_source(handle: Handle<AudioSource>) -> Self {
```

Comment:
Really, because of type inference. I subjectively consider it less ugly to have a separate method like this, than to force users to type out generic type parameters with turbofish syntax.

Because `AssetServer::load` returns a `Handle<T>`, expecting the compiler to infer the `T` from later usage, if then the handle is used with a method that is also generic, the compiler has no way of inferring the type.

If the two methods were merged into one (let's call it `from_source`, equivalent to `from_custom_source`, with the generic type parameter), we would have this problem:

```rust
// this does not work, the compiler gives an error that the type cannot be inferred
AudioBundle::from_source(asset_server.load("thing.wav"));

// and this does not work either (it's the same thing split into two lines)
let handle = asset_server.load("thing.wav");
AudioBundle::from_source(handle);

// unless you have some other usage that locks in the type inference
commands.insert_resource(MyAudioAssetsResource {
    thing: handle.clone(), // where `thing: Handle<AssetSource>`
});

// you would have to do this
AudioBundle::from_source::<AudioSource>(asset_server.load("thing.wav"));
// or this
AudioBundle::from_source(asset_server.load::<AudioSource>("thing.wav"));
// or this
let handle: Handle<AudioSource> = asset_server.load("thing.wav");
```

I find all of those to be very ugly and bad UX. Having to remember to type obtuse boilerplate is not fun. Having a dedicated method for Bevy's default/standard audio asset type seems much more user-friendly.

That said, I could rename `from_audio_source` to `from_source` and keep `from_custom_source` (the opposite of the example above), to make the common case even shorter.